tests/integration/src/test/scala/mesosphere/marathon/integration/AppDeployIntegrationTest.scala

```diff
@@ -95,14 +154,32 @@ class AppDeployIntegrationTest extends AkkaIntegrationTest with EmbeddedMarathon
       waitForStatusUpdates("TASK_RUNNING", "TASK_FAILED")
     }
 
-    def createAFailingAppResultingInBackOff(id: PathId): App = {
-      Given("a new app")
+    def verifyAFailingAppResultingInBackOff(id: PathId): App = {
+      Given("a crash looping app")
```

Comment:
I'd much rather our Given's be in the test than in this method. The Givens get logged with a line number and if there are multiple tests calling this method then it will not be as clear with which test the Given is associated.

I'd say perhaps we call this method "backedOffApp", which creates the app and waits for it to be backed off, but does not do any Given / And. Then we just put "Given a backed off app named X" in the tests themselves.

I understand a desire to be DRY and sometimes that work against us in tests and make things harder to debug or follow. A little repetition is fine.

Reply:
Couldn't agree more :) 

When I broke down the existing methods, I thought to retain the behavior but agree this is better!